server/src/test/java/io/crate/action/sql/CursorTest.java

```diff
@@ -144,4 +144,183 @@ public void test_fetching_absolute_exceeding_last_row() throws Exception {
                 .isExactlyInstanceOf(IllegalArgumentException.class)
                 .hasMessage("Cannot return row: 6, total rows: 5");
     }
+
+    @Test
+    public void test_fetch_backward_all() throws Exception {
+        BatchIterator<Row> rows = TestingBatchIterators.range(1, 6);
+        Cursor cursor = new Cursor(
+                new NoopCircuitBreaker("dummy"),
+                true,
+                Hold.WITHOUT,
+                CompletableFuture.completedFuture(rows),
+                List.of(new InputColumn(0, DataTypes.INTEGER))
+        );
+
+        // Move forward to last row
+        TestingRowConsumer consumer = new TestingRowConsumer();
+        cursor.fetch(consumer, ScrollMode.RELATIVE, 5);
+        assertThat(TestingHelpers.printedTable(consumer.getBucket())).isEqualTo(
```

Comment:
Could be worth adding a `assertThat(RowConsumer).hasRows(String ... row)` as some point. But better to do as follow up in a separate PR.

Reply:
That would be great yes, I have it in mind long time now, since we introduced assertJ.